das/options.go

```diff
@@ -0,0 +1,142 @@
+package das
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+var (
+	ErrInvalidOption = fmt.Errorf("das: invalid option")
+)
+
+func errInvalidOptionValue(optionName string, value string) error {
+	return fmt.Errorf("%w: value %s cannot be %s", ErrInvalidOption, optionName, value)
+}
+
+// Type Option is the functional option that is applied to the daser instance for parameters configuration.
+type Option func(*DASer) error
+
+// Type parameters is the set of parameters that must be configured for the daser
+type Parameters struct {
+	//  SamplingRange is the maximum amount of headers processed in one job.
+	SamplingRange uint64
+
+	// ConcurrencyLimit defines the maximum amount of sampling workers running in parallel.
+	ConcurrencyLimit int
+
+	// BackgroundStoreInterval is the period of time for background checkpointStore to perform a checkpoint backup.
+	BackgroundStoreInterval time.Duration
+
+	// PriorityQueueSize defines the size limit of the priority queue
+	PriorityQueueSize int
+
+	// SampleFrom is the height sampling will start from
+	SampleFrom uint64
+}
+
+// DefaultParameters returns the default configuration values for the daser parameters
+func DefaultParameters() Parameters {
+	// TODO(@derrandz): parameters needs performance testing on real network to define optimal values
+	return Parameters{
+		SamplingRange:           100,
+		ConcurrencyLimit:        16,
+		BackgroundStoreInterval: 10 * time.Minute,
+		PriorityQueueSize:       16 * 4,
+		SampleFrom:              1,
+	}
+}
+
+// Validate validates the valuea in Parameters
```

Comment:
```suggestion
// Validate validates the values in Parameters
```

also refer to my comment on das.Config pls 🙏🏻